lib/valueflow.cpp

```diff
@@ -2501,49 +2506,64 @@ static void valueFlowAfterCondition(TokenList *tokenlist, SymbolDatabase* symbol
-                    if (values.size() == 1U && Token::Match(tok, "==|!")) {
-                        const Token *parent = tok->astParent();
-                        while (parent && parent->str() == "&&")
-                            parent = parent->astParent();
-                        if (parent && parent->str() == "(")
-                            values.front().setKnown();
-                    }
-                    if (!valueFlowForward(startToken->next(), startToken->link(), var, varid, values, true, false, tokenlist, errorLogger, settings))
-                        continue;
-                    values.front().setPossible();
-                    if (isVariableChanged(startToken, startToken->link(), varid, var->isGlobal(), settings)) {
-                        // TODO: The endToken should not be startToken->link() in the valueFlowForward call
-                        if (settings->debugwarnings)
-                            bailout(tokenlist, errorLogger, startToken->link(), "valueFlowAfterCondition: " + var->name() + " is changed in conditional block");
-                        continue;
+                    if (check_if && Token::simpleMatch(top->link(), ") {"))
+                        startTokens[0] = top->link()->next();
+                    if (check_else && Token::simpleMatch(top->link()->linkAt(1), "} else {"))
+                        startTokens[1] = top->link()->linkAt(1)->tokAt(2);
+                }
+
+                int bail = 0;
+                if(check_if && check_else)
+                    bail = 1;
+
+                for(int i=0;i<2;i++) {
+                    Token * startToken = startTokens[i];
+                    if(startToken) {
+                        std::list<ValueFlow::Value> & values = (i==0 ? true_values : false_values);
+                        if (values.size() == 1U && Token::Match(tok, "==|!")) {
+                            const Token *parent = tok->astParent();
+                            while (parent && parent->str() == "&&")
+                                parent = parent->astParent();
+                            if (parent && parent->str() == "(")
+                                values.front().setKnown();
+                        }
+                        if (!valueFlowForward(startTokens[i]->next(), startTokens[i]->link(), var, varid, values, true, false, tokenlist, errorLogger, settings))
+                            continue;
+                        values.front().setPossible();
+                        if (isVariableChanged(startTokens[i], startTokens[i]->link(), varid, var->isGlobal(), settings)) {
+                            // TODO: The endToken should not be startTokens[i]->link() in the valueFlowForward call
+                            if (settings->debugwarnings)
+                                bailout(tokenlist, errorLogger, startTokens[i]->link(), "valueFlowAfterCondition: " + var->name() + " is changed in conditional block");
+                            bail = -1;
```

Comment:
so if you bailout if the "if" then the "else" is not handled neither.

Reply:
Yes, if the variable is changed in the then it wont try running valueflow on else part. This probably needs to be improved.